lib/pbench/cli/agent/commands/tools/list.py

```diff
@@ -107,17 +104,17 @@ def execute(self) -> int:
                 self.logger.error(msg)
                 return 1
 
-        # no tool name was specified
-        if tool_info:
-            found = self.print_results(tool_info, self.context.with_option)
-            if not found:
-                msg = "No tools found"
-                if self.context.group:
-                    msg += f' in group "{self.context.group[0]}"'
-                self.logger.warn(msg)
         else:
-            self.logger.warn("No tool groups found")
-        return 0
+            if tool_info:
+                found = self.print_results(tool_info, self.context.with_option)
+                if not found:
+                    msg = "No tools found"
+                    if self.context.group:
+                        msg += f' in group "{self.context.group[0]}"'
+                    self.logger.warn(msg)
+            else:
+                self.logger.warn("No tool groups found")
```

Comment:
Since we never fall out of the `if` above (we return `0` or `1` based on the nested `if`), and this `else` always drops out, I think this nested `if/else` could be pulled out a level as an `elif` followed by an `else`, with an unconditional `return 0` following.